feign/src/main/java/com/baeldung/feign/clients/builder/BookFeignClientBuilder.java

```diff
@@ -0,0 +1,27 @@
+package com.baeldung.feign.clients.builder;
+
+import com.baeldung.feign.header.interceptor.AuthRequestInterceptor;
+
+import feign.Feign;
+import feign.gson.GsonDecoder;
+import feign.gson.GsonEncoder;
+
+
+public class BookFeignClientBuilder {
+
+    public <T> T createClient(Class<T> type, String uri) {
```

Comment:
These methods should probably be `static`. There's nothing `instance`ey about this code. We don't access the builder via an interface and the methods don't use any fields of the builder class.

Also, I'm surprised by the use of the Gson library here in that Jackson is more popular. However, it's not relevant to the meat of the article.

Reply:
Yes. I have changed to static now.  for Gson, i have reused this module so Gson was already used in this module.  But yes Jackson is what i have alos used in all other instances .